tests/model_fields/test_generatedfield.py

```diff
@@ -16,6 +21,31 @@
 )
 
 
+@isolate_apps("model_fields")
+class GeneratedFieldTransactionTests(TransactionTestCase):
+    available_apps = ["model_fields"]
+
+    def test_generated_sql(self):
+        """
+        Regression test for #34861
+
+        GeneratedField.generated_sql shouldn't generate broken SQL code.
+        """
+        from django.db.models import JSONField
+        from django.db.models.fields.json import KT
+
+        class Response(Model):
+            data = JSONField(default=dict)
+            status = GeneratedField(
+                db_persist=True,
+                expression=KT("data__status"),
+            )
+
+        with connection.schema_editor(collect_sql=True) as editor:
+            editor.create_model(Response)
+            self.assertNotIn("None", "\n".join(editor.collected_sql))
```

Comment:
I'd move it to the `tests/schema/tests.py`.

Reply:
Ok, I'll do that.